src/translation.rs

```diff
@@ -220,6 +220,15 @@ impl<T: Copy, Src, Dst> Translation2D<T, Src, Dst> {
         point2(p.x + self.x, p.y + self.y)
     }
 
+    /// Translate a point by inverse and cast its unit.
+    #[inline]
+    pub(crate) fn transform_point_inv(&self, p: Point2D<T, Dst>) -> Point2D<T::Output, Src>
```

Comment:
Let's not add `_inv` methods like this. You could relax the requirements on `inverse` or simply do the math manually in `sub`.

Reply:
These functions are `pub(crate)` since I wasn't sure they made sense in the public API or what to name them. Having them alongside the transformations in the opposite direction make it easier to see that they're correct, but the body of the function could definitely be moved directly into the `Sub` implementation.

Reply:
> but the body of the function could definitely be moved directly into the Sub implementation.

This would be my preference.